atlp-rwanda/alpha-ec-be
# Ch configure es lint 18730013

#### What does this PR do?

This PR configures ESLint and a Git pre-commit hook to run tests and format the code base according to ESLint and Prettier rules. This ensures that all code committed to the repository adheres to our coding standards and passes all tests.

#### Description of Task to be completed?

- Install ESLint and Prettier.
- Configure ESLint and Prettier according to project standards.
- Install Husky for Git hooks.
- Configure a pre-commit hook to run ESLint and tests.

#### How should this be manually tested?

1. Make a change to the code that should trigger an ESLint error or a failing test.
2. Attempt to commit the change. The commit should be blocked, and the error messages should be displayed.
3. Fix the ESLint error.
4. Attempt to commit the change again. The commit should now succeed.

#### Any background context you want to provide?

This setup is crucial for maintaining code quality and preventing breaking changes from being committed. It ensures that all code adheres to our coding standards and passes all tests before it is merged into the main branch.

#### What are the relevant pivotal tracker/Trello stories?

- Story ID: 187300132
